### ISSUE TITLE
Does the binding controller need to watch the cfapp resource

### ISSUE THREAD

**Issue by georgethebeatle** [cloudfoundry/korifi#3527]
### Background

While introducing support for service bindings to managed service instances we spotted two things

- The binding controller is currently [watching](https://github.com/cloudfoundry/korifi/blob/8b2cf15119b3e44cb03160541aa5ab172bb0c42a/controllers/controllers/services/bindings/controller.go#L77-L80) the cfapp resource

- In the reconcile method of the binding controller we are getting the cfapp and do not do anything with the object (other than returning an error if the get operation fails)

### Action to take

- Find out if this behaviour is really needed and remove it if not.
- If the app lookup is needed just to validate that the app is there we could just get it without assigning it to a local var to make it clear that the get is intentional and that we do not need to look at the result

```
r.k8sClient.Get(ctx, types.NamespacedName{Name: cfServiceBinding.Spec.AppRef.Name, Namespace: cfServiceBinding.Namespace}, new(korifiv1alpha1.CFApp))
```

### Dev notes

- One reason to watch apps from the binding controller might be to allow us to make a binding unready once its app goes away
- Even if the avove is true do we still need to lookup the app? Wouldn't the servicebinding.io binding beome unready if the statefulset goes away? In this case the cfServiceBinding will also go into a bad state

**Comment by danail-branekov** [comment]
There does not seem to have any logical reason for the binding controller to watch apps. The current implementation just retrurns an error if the app is not found which results into the binding not becoming ready. 

Generally, CFApp finalization takes care to [delete all the bindings](https://github.com/cloudfoundry/korifi/blob/df9925de6b12eeb51cc05d3dd70019250b8e0175/controllers/controllers/workloads/apps/controller.go#L364-L386) to the app being deleted. This means that in an ideal world there should be no bindings that exist after the app is gone.

As app finalization is "racy", e.g. it may not "see" a binding that has been just created as the app controller cache is outdated, there is a slight theorethical possibility to have an orphan binding without an app. In this case, however, the binding would be still not ready as the servicebindingio will be not ready as the app statefulset is not there. 

A theorethical kubectl user might apply a binding yaml that references an app that does not exist, but the binding not getting ready would be covered by servicebindingio not becoming ready.

TL;DR, we believe that there is no reason for the binding controller to watch apps. We are going to remove the watch.
